core/src/iam/verify.rs

```diff
@@ -641,6 +633,51 @@ fn verify_pass(pass: &str, hash: &str) -> Result<(), Error> {
 	}
 }
 
+// Execute the AUTHENTICATE clause for a record access method
+async fn authenticate_record(
+	kvs: &Datastore,
+	session: &Session,
+	authenticate: Value,
+) -> Result<Thing, Error> {
+	match kvs.evaluate(authenticate, session, None).await {
+		Ok(val) => match val.record() {
+			// If the AUTHENTICATE clause returns a record, authentication continues with that record
+			Some(id) => Ok(id),
+			// If the AUTHENTICATE clause returns anything else, authentication fails generically
+			_ => Err(Error::InvalidAuth),
+		},
+		Err(e) => match e {
+			// If the AUTHENTICATE clause throws a specific error, authentication fails with that error
+			Error::Thrown(_) => Err(e),
+			e if *INSECURE_FORWARD_RECORD_ACCESS_ERRORS => Err(e),
+			_ => Err(Error::InvalidAuth),
+		},
+	}
+}
+
+// Execute the AUTHENTICATE clause for a JWT access method
+async fn authenticate_jwt(
```

Comment:
It will also cover bearer access right?
```suggestion
async fn authenticate_system(
```

Reply:
I am not sure how this will fit with bearer access, so I would prefer to make this change in #4302 if necessary.

Reply:
I mean that maybe the `AUTHENTICATE` clause will have a specific implementation for bearer access or maybe you are right and we will call the exact same function and should rename it to a more generic name.